### ISSUE TITLE
Installed header files with name conflicts

### ISSUE THREAD

**Issue by sebastiangrimberg** [hypre-space/hypre#898]
When installing Hypre, some header files with fairly generic names are installed to the top level `INSTALL_PREFIX/include` directory which could lead to conflicts. For example `fortran.h` and `krylov.h`.

Would it be possible to rename these files to have some prefix as most of the other Hypre headers do, or to instead install them to a subdirectory at `INSTALL_DIR/include/hypre`? Or perhaps there's another preferred approach for avoiding these conflicts that could be taken. Thank you very much for your help!

**Comment by rfalgout** [comment]
Hi @sebastiangrimberg .  This is definitely something we need to fix.  It's been on the back burner for a while now because nobody has really had any issues so far.  Sounds like we need to move it up in priority.  Thanks!

**Comment by vdilecce** [comment]
I also agree that it would be a much more robust approach to include HYPRE headers as `<hypre/HYPRE_parcsr_ls.h>` instead of `<HYPRE_parcsr_ls.h>`.

That would also be of help when `.clang-format`ing files making it possible to define one include category regex only for grouping (e.g., `"<hypre/.*>"`).
